src/Microsoft.DncEng.SecretManager/SecretManifest.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Text;
+using YamlDotNet.Core;
+using YamlDotNet.Serialization;
+
+namespace Microsoft.DncEng.SecretManager
+{
+    public class SecretManifest
+    {
+        public static SecretManifest Read(string filePath)
+        {
+            using var reader = new StreamReader(filePath, Encoding.UTF8);
+            return Parse(reader);
+        }
+
+        public static SecretManifest Parse(TextReader reader)
+        {
+            IDeserializer deserializer = new DeserializerBuilder()
+                .Build();
+
+            var parser = new Parser(reader);
+            return new SecretManifest(deserializer.Deserialize<Format>(parser));
+        }
+
+        #pragma warning disable IDE1006
+        // ReSharper disable All
+        private class Format
+        {
+            public Guid subscription { get; set; }
+            public string name { get; set; }
+            public bool missingSecretsAllowed { get; set; }
+            public Dictionary<string, Key> keys { get; set; }
+            public Dictionary<string, Secret> secrets { get; set; }
+
+            public class Key
+            {
+                public string type { get; set; }
+                public int size { get; set; }
+            }
+
+            public class Secret
+            {
+                public string type { get; set; }
+                public Dictionary<string, string> parameters { get; set; }
+            }
+        }
+        #pragma warning restore IDE1006
+        // ReSharper restore All
+
+        private SecretManifest(Format data)
+        {
+            Subscription = data.subscription;
+            Name = data.name;
+            MissingSecretsAllowed = data.missingSecretsAllowed;
+            Keys = data.keys.ToImmutableDictionary(p => p.Key, p => CreateKey(p.Value));
+            Secrets = data.secrets.ToImmutableDictionary(p => p.Key, p => CreateSecret(p.Value));
+        }
+
+        public Guid Subscription { get; }
+        public string Name { get; }
```

Comment:
how do you intend to handle things like "owner" and "description"?

Reply:
Ahh yea, those should be in here. I didn't have them in the stuff I was looking at I will add them.